File: 0700-search-in-a-binary-search-tree/0700-search-in-a-binary-search-tree.c

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

struct TreeNode* searchBST(struct TreeNode* root, int val){

    if (!root || root->val == val)
        return root;
    if (root->val > val)
     root = searchBST(root->left,val);
    else
      root = searchBST(root->right,val);
    return root;
}

File: 0700-search-in-a-binary-search-tree/0700-search-in-a-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var searchBST = function(root, val) {
    while(root)
    {
        if (root.val == val) return root;
        else if(root.val < val)
           root = searchBST(root.right,val);
        else root = searchBST(root.left,val); 
    }
    return null;
};